HVSample/src/main/java/com/microsoft/hsg/android/hvsample/ActionPlanActivity.java

```diff
@@ -0,0 +1,135 @@
+package com.microsoft.hsg.android.hvsample;
+
+import java.util.ArrayList;
+import java.util.List;
+import java.util.Observable;
+import java.util.concurrent.Callable;
+import java.util.concurrent.ExecutorService;
+import java.util.concurrent.Executors;
+
+import com.microsoft.hsg.Connection;
+import com.microsoft.hsg.HVException;
+import com.microsoft.hsg.android.simplexml.HealthVaultApp;
+import com.microsoft.hsg.android.simplexml.HealthVaultSettings;
+import com.microsoft.hsg.android.simplexml.ShellActivity;
+import com.microsoft.hsg.android.simplexml.client.HealthVaultClient;
+import com.microsoft.hsg.android.simplexml.client.HealthVaultRestClient;
+import com.microsoft.hsg.android.simplexml.client.RequestCallback;
+import com.microsoft.hsg.android.simplexml.methods.getthings3.request.ThingRequestGroup2;
+import com.microsoft.hsg.android.simplexml.methods.getthings3.response.ThingResponseGroup2;
+import com.microsoft.hsg.android.simplexml.things.thing.Thing2;
+import com.microsoft.hsg.android.simplexml.things.types.types.PersonInfo;
+import com.microsoft.hsg.android.simplexml.things.types.types.Record;
+import com.microsoft.hsg.android.simplexml.things.types.weight.Weight;
+
+import android.app.Activity;
+import android.content.Intent;
+import android.os.Bundle;
+import android.util.Log;
+import android.view.View;
+import android.view.Window;
+import android.widget.AdapterView;
+import android.widget.ArrayAdapter;
+import android.widget.Button;
+import android.widget.EditText;
+import android.widget.ListView;
+import android.widget.Spinner;
+import android.widget.Toast;
+import android.widget.AdapterView.OnItemSelectedListener;
+
+import healthvault.client.implementation.MicrosoftHealthVaultRestApiImpl;
+import healthvault.client.models.ActionPlan;
+import healthvault.client.models.ActionPlansResponseActionPlanInstance;
+
+import healthvault.client.models.Objective;
+import okhttp3.ResponseBody;
+import retrofit2.Response;
+import rx.Subscriber;
+import rx.Subscription;
+import rx.schedulers.Schedulers;
+
+public class ActionPlanActivity  extends Activity {
+	private HealthVaultApp mService;
+	private HealthVaultClient mClient;
+	private Record mCurrentRecord;
+	private static ActionPlansResponseActionPlanInstance mActionPlanInstance;
+
+	@Override
+	public void onCreate(Bundle savedInstanceState) {
+		super.onCreate(savedInstanceState);
+		setContentView(R.layout.activity_actionplan);
+		mService = HealthVaultApp.getInstance();
+		mClient = new HealthVaultClient();
+
+		if (mService.isAppConnected()) {
+			mCurrentRecord = HealthVaultApp.getInstance().getCurrentRecord();
```

Comment:
Left the logic in the OnCreate

---
In reply to: [118065595](https://github.com/Microsoft/healthvault-java-sdk/pull/5#discussion_r118065595) [](ancestors = 118065595)